src/GuzzleMiddlewareFactory.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry;
+
+use Exception;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+use Sentry\SentrySdk;
+
+class GuzzleMiddlewareFactory
+{
+    private const MAX_LOG_BODY_IN_CHARS = 200;
+
+    public function __construct(private LoggerInterface $logger)
+    {
+    }
+
+    public function factory(callable $handler): callable
+    {
+        return function (RequestInterface $srcRequest, array $options) use ($handler) {
+            $traceHeader = SentrySdk::getCurrentHub()->getSpan()?->toTraceparent();
+            if ($traceHeader) {
+                $request = $srcRequest->withAddedHeader('sentry-trace', $traceHeader);
+            } else {
+                $request = clone $srcRequest;
+            }
+
+            $startTime = microtime(true);
+            $path = $request->getMethod() . ':' . $request->getUri()->__toString();
+            $requestBody = $request->getBody()->isReadable()
+                ? $request->getBody()->getContents()
+                : '[not readable]';
+            if (mb_strlen($requestBody) > self::MAX_LOG_BODY_IN_CHARS) {
```

Comment:
Could be simplified with `mb_strimwidth()`. https://www.php.net/manual/ru/function.mb-strimwidth.php